sdk/resources/mgmt/src/main/java/com/azure/management/resources/fluentcore/authentication/AzureTokenCredential.java

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.management.resources.fluentcore.authentication;
+
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.management.AzureEnvironment;
+
+/**
+ * A credential provider for Azure authentication
+ * before using the APIs provided in the library.
+ */
+public class AzureTokenCredential {
+
+    private String domain;
```

Comment:
how about tenant for multiple-tenant?

Reply:
@yungezz multi-tenant should use `AuxiliaryAuthenticationPolicy`. https://github.com/azure/azure-sdk-for-java/blob/036b9f6e6ff54a0e497f1b433f67f0eb4c32625d/sdk/resources/mgmt/src/main/java/com/azure/management/resources/fluentcore/policy/AuxiliaryAuthenticationPolicy.java#L27